src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -99,20 +99,40 @@ def filter_step(self, start, stop, randvar, data, **kwargs):
         """
         data = np.asarray(data)
         predrv, _ = self.predict(start, stop, randvar)
-        filtrv, _, _, _ = self.update(stop, predrv, data)
+        filtrv, _, _ = self.update(stop, predrv, data)
         return filtrv
 
     def predict(self, start, stop, randvar, **kwargs):
         return self.dynamod.transition_rv(randvar, start, stop=stop, **kwargs)
 
     def update(self, time, randvar, data, **kwargs):
+        """
```

Comment:
I know I'm annoying, but I think since you already started adding a docstring it might make sense to add a one-liner for the title, types for parameters and parameter descriptions.

What about:
"Update the state by conditioning on data"

Function signature with type hints:
`def update(self, time: FloatArgType, randvar: RandomVariable, data:np.ndarray, **kwargs):` (no idea how to handle kwargs.

Parameter description:
```
        Parameters
        ----------
        time
            Time at which the measurement model should be evaluated
        randvar
            State 
        data
            True, noiseless measurement 
```

Something like that?

Reply:
Shouldn't have started a docstring...